R/compiler.R

```diff
@@ -137,6 +137,12 @@ SubstraitCompiler <- R6::R6Class(
     #'
     #' @return `self`
     validate = function() {
+
+      # DuckDB backend doesn't accept empty SELECT clause
+      if (inherits(self, "DuckDBSubstraitCompiler") && length(self$schema$names) == 0) {
+        rlang::abort("Column list must not be empty")
```

Comment:
Feel free to add a suggestion, this was a "least worst than the default error message" change to prevent lots of extraneous output.  At this stage in the package development, the focus is on getting the basics done, but always happy for tweaks/PRs if you have suggestions here.